src/com/georgidinov/domain/record/BorrowRequest.java

```diff
@@ -0,0 +1,76 @@
+package com.georgidinov.domain.record;
+
+import com.georgidinov.domain.book.Book;
+import com.georgidinov.domain.person.Person;
+
+import java.time.LocalDate;
+
+import static com.georgidinov.util.MyDateFormatter.getDateString;
+
+public class BorrowRequest {
+
+    public static final int ACTIVE_RESERVATION_PERIOD_IN_DAYS = 3;
+
+    private static int instanceCounter = 0;
+    private int id;
+
+    private int bookId;
+    private int readerId;
+    private LocalDate reservationDueDate;
+    private boolean isActive;
+
+    public BorrowRequest(Book book, Person person) {
+        this.bookId = book.getId();
+        this.readerId = person.getId();
+        this.reservationDueDate = LocalDate.now().plusDays(ACTIVE_RESERVATION_PERIOD_IN_DAYS);
+        this.isActive = true;
+        this.id = ++instanceCounter;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public int getBookId() {
+        return bookId;
+    }
+
+    public int getReaderId() {
+        return readerId;
+    }
+
+    public LocalDate getReservationDueDate() {
+        return reservationDueDate;
+    }
+
+    public String getReservationDueDateString() {
+        return getDateString(this.reservationDueDate);
+    }
+
+    public boolean isActive() {
+        return isActive;
+    }
+
+    public void changeState() {
+        if (this.reservationDueDate.isEqual(LocalDate.now())) {
```

Comment:
I didn't check the logic where this method is used, but `changeState` sounds like I will change something . But if the condition is false, then I won't understand that nothing has been changed